gufe/custom_codecs.py

```diff
@@ -33,4 +71,36 @@
     ).reshape(dct['shape'])
 )
 
+SETTINGS_CODEC = JSONCodec(
+    cls=SettingsBaseModel,
+    to_dict=lambda obj: {field: getattr(obj, field) for field in obj.__fields__},
+    from_dict=default_from_dict,
+    is_my_dict=functools.partial(inherited_is_my_dict, cls=SettingsBaseModel),
+)
+
+OPENFF_QUANTITY_CODEC = JSONCodec(
+    cls=None,
+    to_dict=lambda obj: {
+        "magnitude": obj.m,
+        "unit": str(obj.u),
+        ":is_custom:": True,
+        "pint_unit_registry": "openff_units",
+    },
+    from_dict=lambda dct: openff.units.DEFAULT_UNIT_REGISTRY(
```

Comment:
drop `openff.units`, since we've already imported `DEFAULT_UNIT_REGISTRY` (allows us to drop `openff.units` import, as well)